<div align="center">

<h1 style="font-size: 18px;"> <b>LAPORAN PRAKTIKUM</b></h1>
<h2 style="font-size: 18px;"> <b>ALGORITMA DAN PEMROGRAMAN</b></h2>
<h2 style="font-size: 18px;"> <b>"Pengenalan Variabel dan Tipe Data Python"</b></h2>
<h4 style="font-size: 16px;"> <b>Disusun Oleh: </b></h4>
<h4 style="font-size: 16px; "> <b>Muhammad Rasya Fahrozi</b> </h4>
<h4 style="font-size: 16px; "> <b>2611531007</b><h4>
<h4 style="font-size: 16px;"> <b>Dosen Pengampu:</b></h4>
<h4 style="font-size: 16px; "> <b>Dr. Ir. Wahyudi, S.T., M.T.</b> </h4>
<h4 style="font-size: 16px;"> <b>Asisten Praktikum: </b></h4>
<h4 style="font-size: 16px;"> <b>Muhammad Galid Avero</b></h4>

<br>

 <img src="images/logo_unand.png" width="200">

<br>

<h4 style="font-size: 16px;"> <b>FAKULTAS TEKNOLOGI INFORMASI</b></h4>
<h4 style="font-size: 16px; "> <b>DEPARTEMEN INFORMATIKA</b> </h4>
<h4 style="font-size: 16px;"> <b>UNIVERSITAS ANDALAS </b></h4>
<h4 style="font-size: 16px;"> <b>2026</b></h4>

</div>

## <h1 align = "center">KATA PENGANTAR <h1>

Puji syukur penulis panjatkan atas kehadirat Tuhan Yang Maha Esa yang telah memberikan rahmat dan berkatnya, sehingga penulis dapat menyelesaikan laporan pratikum pengenalan pyhton, vs code, dan koneksi ke github yang telah dilaksanakan sebagai pertemuan ke-1 untuk mata kuliah Praktikum Algoritma dan Pemrograman. Praktikum ini berfokus pada pembahasan pengenalan pyhton, vs code, dan koneksi ke github.

Melalui praktikum ini, hasil yang diharapkan adalah penulis dapat memahami dan mengimplementasikan secara langsung konsep dasar pemrograman Python pada real project.

Penulis menyadari bahwa laporan praktikum ini masih memiliki banyak kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan demi perbaikan dan pengembangan di masa mendatang.

Padang, 2026

Hormat Saya

(Muhammad Rasya Fahrozi)

---
# <h1 align = "center">BAB I <h1>
# <h1 align = "center">PENDAHULUAN<h1>

## 1.1 Latar Belakang
Perkembangan teknologi informasi yang semakin pesat saat ini menuntut setiap individu, khususnya mahasiswa di bidang teknologi, untuk memiliki kemampuan dasar dalam pemrograman komputer. Kemampuan ini menjadi salah satu keterampilan penting yang harus dikuasai, mengingat hampir seluruh aspek kehidupan saat ini telah bertransformasi ke arah digital, mulai dari sistem informasi, pengolahan data, hingga pengembangan aplikasi berbasis perangkat lunak. Untuk dapat menguasai keterampilan tersebut, diperlukan pemahaman terhadap konsep dasar pemrograman terlebih dahulu, salah satunya adalah bahasa pemrograman Python.

Python merupakan salah satu bahasa pemrograman tingkat tinggi yang banyak digunakan saat ini karena sintaksnya yang sederhana, mudah dipahami, serta bersifat open-source sehingga dapat dikembangkan secara bebas oleh siapa pun. Python juga dikenal sebagai bahasa pemrograman yang bersifat dynamically typed, di mana penentuan tipe data suatu variabel dilakukan secara otomatis oleh interpreter berdasarkan nilai yang diberikan, tanpa perlu dideklarasikan secara eksplisit oleh pengguna. Karakteristik inilah yang menjadikan Python sebagai salah satu bahasa pemrograman yang direkomendasikan bagi pemula yang baru memulai proses belajar pemrograman.

Salah satu konsep paling mendasar yang harus dipahami dalam mempelajari suatu bahasa pemrograman adalah variabel dan tipe data. Variabel berfungsi sebagai tempat penyimpanan nilai atau data yang dapat digunakan dan diproses kembali dalam suatu program, sedangkan tipe data menentukan jenis nilai yang dapat disimpan dalam variabel tersebut, seperti bilangan bulat (integer), bilangan desimal (float), teks (string), maupun nilai logika (boolean). Selain itu, pemahaman mengenai konstanta serta fungsi input() untuk menerima masukan dari pengguna juga menjadi bagian penting yang mendukung penyusunan program yang lebih interaktif dan dinamis.

Oleh karena itu, melalui praktikum ini, mahasiswa diharapkan dapat memahami secara langsung konsep variabel, tipe data, konstanta, serta penggunaan fungsi input() dalam bahasa pemrograman Python, sehingga dapat menjadi bekal dasar dalam mempelajari materi-materi pemrograman yang lebih lanjut pada tahap berikutnya.

## 1.2 Tujuan
1. Memahami konsep dasar bahasa pemrograman Python, seperti apa itu variabel dan aturan aturan dalam pembuatan variabel.
2. Mengetahui jenis-jenis tipe data dasar dalam Python, seperti integer, float, string, dan boolean.
3. Memahami cara penggunaan fungsi type() untuk memeriksa tipe data suatu variabel.

## 1.3 Manfaat
1. Melatih kemampuan berpikir kritis dalam penggunaan setiap tipe data pada program sesuai dengan fungsinya.
2. Menjelaskan tiap program yang dibuat dari setiap tipe data.
3. Memperoleh pengalaman langsung dalam menyelesaikan permasalahan teknis.


---
# <h1 align = "center">BAB II <h1> 
## <h1 align = "center">PEMBAHASAN <h1>

## 2.1 Dasar Teori
### 2.1.1 Variabel
Variabel merupakan lokasi penyimpanan yang digunakan untuk menyimpan suatu nilai atau data dalam program, dan ketika sebuah variabel didefinisikan maka Python secara otomatis mengalokasikan ruang di memori untuk menyimpan nilai tersebut. Berbeda dengan sejumlah bahasa pemrograman lain, Python tidak mengharuskan programmer mendeklarasikan tipe data secara eksplisit saat membuat variabel; cukup dengan menuliskan nama variabel beserta nilainya, dan Python akan menentukan tipe datanya secara otomatis. Pendekatan ini dikenal dengan istilah dynamically typed, yang berarti tipe data suatu variabel dapat berubah secara dinamis selama proses eksekusi program berlangsung.

### 2.1.2 Numeric
Python secara otomatis menentukan tipe data suatu variabel berdasarkan nilai yang diberikan kepadanya. Tipe data numerik pada Python terdiri atas tipe int, yaitu tipe data untuk menyimpan bilangan bulat tanpa nilai desimal, tipe float untuk menyimpan bilangan berkoma (floating point), serta tipe complex untuk menyimpan bilangan kompleks dengan format a + bj, di mana a merupakan bagian real dan b merupakan bagian imajiner . Secara garis besar, jenis tipe data angka pada Python terbagi menjadi integer (bilangan bulat) dan float (bilangan pecahan).

### 2.1.3 String
String atau str adalah tipe data yang digunakan untuk menyimpan nilai dalam bentuk teks, yang penulisannya dapat diapit menggunakan tanda kutip ganda maupun tanda kutip tunggal. Dalam praktiknya, data string sering dikombinasikan dengan variabel dari tipe data lain, dan terdapat beberapa cara untuk memformat hasil kombinasi tersebut, yaitu menggunakan operator %, fungsi format(), maupun f-string, di mana f-string saat ini lebih disarankan karena sintaksnya lebih ringkas dan mudah dipahami.

### 2.1.4 Boolean
Tipe data boolean merupakan tipe data yang hanya memiliki nilai benar atau salah, dan pada Python jenis tipe data ini termasuk ke dalam kategori tipe data dasar selain tipe data angka dan tipe data teks. Contoh penerapannya dapat dilihat pada deklarasi variabel dengan nilai True atau False, sebagaimana pada tipe data lain yang jenisnya ditentukan otomatis oleh Python berdasarkan nilai yang diberikan.

### 2.1.5 Konstanta
Dalam bahasa pemrograman Python, penulisan konstanta pada dasarnya mengikuti aturan penulisan variabel pada umumnya, namun secara konvensi nilai yang dimaksudkan sebagai konstanta dituliskan menggunakan huruf kapital untuk menandakan bahwa nilainya tidak dimaksudkan untuk diubah selama program berjalan. Salah satu contoh nya adalah nilai PI

### 2.1.6 Input
Fungsi input() merupakan fungsi bawaan (built-in) pada Python yang digunakan untuk menerima masukan dari pengguna melalui keyboard, dan fungsi ini bekerja dengan menerima satu parameter opsional berupa teks yang akan ditampilkan sebagai prompt sebelum pengguna memasukkan data. Nilai yang dimasukkan pengguna melalui fungsi input() akan selalu dikembalikan dalam bentuk tipe data string, sehingga apabila nilai tersebut akan digunakan dalam perhitungan matematis, perlu dilakukan konversi terlebih dahulu ke tipe data numerik menggunakan fungsi seperti int() atau float(). Pada versi Python 3, seluruh proses pengambilan input dari keyboard cukup dilakukan menggunakan fungsi input(), karena fungsi raw_input() yang terdapat pada Python 2 telah digabungkan ke dalamnya.

## 2.2 Kode Program

### 2.2.1 Program Numerik "A_2611531007"




In [ ]:
#create a variable with integer value
a_1007 = 100
print("The type of variable having value", a_1007, " is", type (a_1007))
      
#create a variable with float value
b_1007 = 10.2345
print("The type of varible having value", b_1007, " is", type (b_1007))  

#create a variable with complex value
c_1007 = 100+3j
print("The type of variable having value", c_1007, " is", type (c_1007))


The type of variable having value 100  is <class 'int'>
The type of varible having value 10.2345  is <class 'float'>
The type of variable having value (100+3j)  is <class 'complex'>



Penjelasan Program:

1. Baris 1,5 dan 9 adalah komentar (comment)
2. Baris 2 Membuat variabel bernama a_1007 dan mengisinya dengan nilai 100 (tipe data int).
3. Baris 3 Mencetak teks, nilai dari a_1007 (100), dan tipe datanya yang diperiksa menggunakan fungsi type(a_1007) (menghasilkan <class 'int'>).
4. Baris 6 Membuat variabel b_2007 dan mengisinya dengan nilai 10.2345 (tipe data float).
5. Baris 7 Mencetak teks, nilai dari b_1007 (10.2345), dan tipe datanya via type(b_1007) (menghasilkan <class 'float'>).
6. Baris 10 Membuat variabel c_1007 dan mengisinya dengan nilai 100+3j (tipe data complex, di mana 100 adalah bagian riil dan 3j adalah bagian imajiner).
7. Baris 11 Mencetak teks, nilai dari c_1007 ((100+3j)), dan tipe datanya via type(c_1007) (menghasilkan <class 'complex'>).

### 2.2.2 Program String "A_2611531007"

In [ ]:
# Buat nama file dengan nama String_2611532007.py
#nama variable ditambah 4 digit nim terakhir contoh: nama_1234
#ganti programnya dan data masing2
nama_1007 = "Muhammad Rasya Fahrozi"
jenis_kelamin_1007 = 'L'
alamat_1007 = """
    Jl. MR M Roem N0.22 
    Kec. Padang Panjang Timur 
    Kota Padang Panjang 
    """
agama_1007 = 'islam'
print("Nama saya adalah: ", nama_1007)
print("Jenis kelamin saya adalah: ", jenis_kelamin_1007)
print("Alamat saya adalah: ", alamat_1007)
print("Agama saya adalah: ", agama_1007)

Nama saya adalah:  Muhammad Rasya Fahrozi
Jenis kelamin saya adalah:  L
Alamat saya adalah:   
    Jl. MR M Roem N0.22 
    Kec. Padang Panjang Timur 
    Kota Padang Panjang 
    
Agama saya adalah:  islam


Penjelasan Program:

1. Baris 1-3 adalah baris komentar ini cuma catatan/keterangan buat programer, bukan perintah.(#) untuk komentar 1 baris, sedangkan ("""...""") untuk komentar banyak baris.
2. Baris 4 membuat variabel bernama nama_1007 dan mengisinya dengan teks/string "Muhammad Rasya Fahrozi".
3. Baris 5 membuat variabel jenis_kelamin_1007 dan mengisinya dengan karakter 'L'
4. Baris 6 membuat variabel alamat_1007 menggunakan tiga tanda petik ("""). Tiga tanda petik digunakan untuk membuat string banyak baris (multiline string), sehingga format baris baru dan spasi di dalamnya tetap terjaga.
5. Baris 7 membuat variabel agama_1007 dan mengisinya dengan teks/string 'islam'.
6. Baris 8 mencetak kalimat "Nama saya adalah: " diikuti oleh nilai dari variabel nama_1007 (yaitu "Muhammad Rasya Fahrozi").
7. Baris 9 mencetak kalimat "Jenis kelamin saya adalah: " diikuti oleh nilai dari variabel jenis_kelamin_2007 (yaitu 'L').
8. Baris 10 mencetak kalimat "Alamat saya adalah: " diikuti oleh isi variabel alamat_2007 yang mencakup teks multiline.
9. Baris 11 mencetak kalimat "Agama saya adalah: " diikuti oleh nilai dari variabel agama_2007 (yaitu 'islam').

### 2.2.3 Program Boolean "A_2611531007"

In [ ]:
# Buat file dengan nama Boolean_2611532007.py
# Nama variable ditambah 4 digit nim terakhir contoh: nilai_1234
# Deklarasi variable dengan tipe data Boolean
is_lulus_1007 = True
is_cumalude_1007 = True

# Menggunakan Boolean
nilai_1007 = 99
batas_lulus_1007 = 75

# Menentukan nilai Boolean dari kondisi
status_kelulusan_1007 = nilai_1007 >= batas_lulus_1007

print("=== Check Kelulusan ===")
print("Nilai: ", nilai_1007)
print("Apakah Lulus?:", status_kelulusan_1007)
if is_lulus_1007 and is_cumalude_1007:
    print("Selamat, Anda lulus dengan prediket Cum Laude!")

Penjelasan Program:

1. Baris 1-3,7 dan 11 merupakan baris yang komentar
2. Baris 4 Membuat variabel is_lulus_2007 bernilai True (benar).
3. Baris 5 Membuat variabel is_cumlaude_2007 bernilai True (benar).
4. Baris 8 Membuat variabel nilai_2007 bernilai 85 (tipe integer).
5. Baris 9 Membuat variabel batas_lulus_2007 bernilai 75 (tipe integer).
6. Baris 12 Mengevaluasi apakah 85 >= 75. Karena nilainya benar, variabel status_kelulusan_2007 menyimpan nilai Boolean True.
7. Baris 14 Mencetak teks judul "=== Check Kelulusan ===" ke layar.
8. Baris 15 Mencetak label "Nilai:" beserta isi variabel nilai_2007 (85).
9. Baris 16 Mencetak label "Apakah Lulus?:" beserta hasil evaluasi variabel status_kelulusan_2007 (True).
10. Baris 17 Pengondisian menggunakan operator logika and. Memeriksa apakah kedua variabel (is_lulus_2007 DAN is_cumlaude_2007) bernilai True.
11. Baris 18 ini akan dieksekusi dan mencetak teks pesan selamat jika kondisi pada perintah if terpenuhi.

### 2.2.4 Program Input "A_2611531007"

In [3]:
# Buat file dengan nama Input_NIM.py
# Buat program untuk menghitung luas persegi panjang
# Nama variabel ditambah 4 digit nim terakhir contoh: panjang_1234
# Programini menggunakan fungsi input () utnuk meminta pengguna
# memasukkan nilai panjang dan lebar persegi panjang
# Nilai yang dimasukkan akan dokenversi menajdi tipe data integer
 
panjang_1007 = input('Masukkan nilai panjang: ')
lebar_1007 = input ('Masukkan nilai lebar: ')
luas_1007 = int(panjang_1007) * int(lebar_1007)
print("Luas =", luas_1007)

Luas = 12


Penjelasan Program:

1. Baris 1-6 merupakan baris komentar yang berisi intruksi yang akan di jalankan di program ini
2. Baris 7 Menampilkan pesan ke layar dan meminta masukan (input) dari pengguna untuk nilai panjang, lalu menyimpannya ke dalam variabel panjang_1007 (bertipe data string).
3. Baris 8 Menampilkan pesan ke layar dan meminta masukan dari pengguna untuk nilai lebar, lalu menyimpannya ke dalam variabel lebar_1007 (bertipe data string).
4. Baris 9 Mengubah (type casting) isi variabel panjang_1007 dan lebar_1007 yang tadinya teks/string menjadi bilangan bulat dengan fungsi int(), kemudian mengalikan keduanya untuk menghitung luas dan hasilnya disimpan ke variabel luas_1007.
5. Baris 10 Mencetak teks "luas =" beserta nilai hasil perhitungan yang tersimpan pada variabel luas_1007 ke layar.

### 2.2.5 Program Konstanta "A_2611531007"

In [2]:
# Buat file dengan nama Konstanta_1234
# Programini menggunakan konstanta untuk menghitung luas lingkaran
# nama variabel ditambah 4 digit nim terakhir contoh: jari_1234

from typing import Final
PI: Final = 3.14
print("pi: %F" % (PI))
jari_1007 = float(input('Masukkan nilai jari jari: '))
luas_1007 = PI * jari_1007 * jari_1007
print("Luas Lingkaran dengan jari-jari %.2f adalah %.2f" % (jari_1007, luas_1007))

pi: 3.140000
Luas Lingkaran dengan jari-jari 2.00 adalah 12.56


Penjelasan Program:

1. Baris 1-3 merupakan baris komentar yang membarikan intruksi tentang program .
2. Baris 4 Mengimpor penanda tipe (type hint) Final dari modul bawaan typing untuk menegaskan bahwa suatu variabel ditujukan sebagai konstanta (nilainya tidak boleh diubah).
3. Baris 5 Mendeklarasikan konstanta bernama PI bernilai 3.14 dan menandainya menggunakan : Final agar diperlakukan sebagai nilai tetap.
4. Baris 6 Mencetak nilai konstanta PI menggunakan format penulisan string legacy (%f), yang secara default akan menampilkan 6 angka di belakang koma (3.140000). Meminta masukan (input) dari pengguna untuk nilai jari-jari, lalu langsung mengonversi teks masukan tersebut menjadi tipe bilangan desimal menggunakan fungsi float(), dan menyimpannya ke variabel jari_1007.
5. Baris 7 Menghitung luas lingkaran dengan rumus 
 (PI * jari_1007 * jari_1007) dan menyimpannya ke dalam variabel luas_1007.
6. Baris 8 Mencetak hasil ke layar menggunakan format penulisan string (%.2f), di mana %.2f membatasi tampilan angka desimal variabel jari_1007 dan luas_2007 agar hanya memunculkan 2 angka di belakang koma.

### 2.2.6 Program Input "A_2611531007"

In [1]:
# Buat file dengan nama Input_NIM.py
# Buat program untuk menghitung luas persegi panjang
# Nama variabel ditambah 4 digit nim terakhir contoh: panjang_1234
# Programini menggunakan fungsi input () utnuk meminta pengguna
# memasukkan nilai panjang dan lebar persegi panjang
# Nilai yang dimasukkan akan dokenversi menajdi tipe data integer
 
panjang_1007 = input('Masukkan nilai panjang: ')
lebar_1007 = input ('Masukkan nilai lebar: ')
luas_1007 = int(panjang_1007) * int(lebar_1007)
print("Luas =", luas_1007)

Luas = 42


Penjelasan Program:

1. Baris 1-6 merupakan baris komentar yang berisi intruksi yang akan di jalankan di program ini
2. Baris 7 Menampilkan pesan ke layar dan meminta masukan (input) dari pengguna untuk nilai panjang, lalu menyimpannya ke dalam variabel panjang_1007 (bertipe data string).
3. Baris 8 Menampilkan pesan ke layar dan meminta masukan dari pengguna untuk nilai lebar, lalu menyimpannya ke dalam variabel lebar_1007 (bertipe data string).
4. Baris 9 Mengubah (type casting) isi variabel panjang_1007 dan lebar_1007 yang tadinya teks/string menjadi bilangan bulat dengan fungsi int(), kemudian mengalikan keduanya untuk menghitung luas dan hasilnya disimpan ke variabel luas_1007.
5. Baris 10 Mencetak teks "luas =" beserta nilai hasil perhitungan yang tersimpan pada variabel luas_1007 ke layar.

---
# <h1 align = "center">BAB III <h1> 
# <h1 align = "center">KESIMPULAN <h1> 

Berdasarkan hasil praktikum yang telah dilakukan mengenai pengenalan variabel dan tipe data pada bahasa pemrograman Python, dapat disimpulkan bahwa variabel merupakan tempat penyimpanan nilai atau data yang dapat digunakan dan diubah selama program berjalan, tanpa perlu dideklarasikan tipe datanya secara eksplisit, karena Python bersifat dynamically typed sehingga tipe data suatu variabel ditentukan secara otomatis berdasarkan nilai yang diberikan kepadanya.

Selain itu, Python memiliki beberapa tipe data numerik, yaitu integer untuk menyimpan bilangan bulat, float untuk menyimpan bilangan desimal, dan complex untuk menyimpan bilangan kompleks, yang masing-masing memiliki karakteristik dan fungsi penyimpanan nilai yang berbeda. Di samping tipe data numerik, terdapat pula tipe data string yang digunakan untuk menyimpan data berupa teks dan dapat dituliskan menggunakan tanda kutip tunggal maupun ganda, serta tipe data boolean yang hanya memiliki dua kemungkinan nilai, yaitu True dan False, yang umumnya digunakan dalam proses pengambilan keputusan dan operasi logika dalam program.

Adapun konstanta pada Python tidak memiliki mekanisme deklarasi khusus sebagaimana pada beberapa bahasa pemrograman lain, namun secara konvensi dituliskan menggunakan huruf kapital untuk menandakan bahwa nilainya tidak boleh diubah selama program berjalan. Sementara itu, fungsi input() memungkinkan program untuk menerima masukan data dari pengguna melalui keyboard, di mana nilai yang diterima akan selalu berupa tipe data string, sehingga apabila nilai tersebut hendak digunakan dalam operasi numerik, perlu dilakukan konversi tipe data (type casting) terlebih dahulu menggunakan fungsi seperti int() atau float().

Secara keseluruhan, dapat disimpulkan bahwa pemahaman terhadap konsep variabel dan tipe data merupakan fondasi penting dalam pemrograman Python, karena keduanya menjadi dasar utama dalam penyusunan logika program yang lebih kompleks pada tahap pembelajaran selanjutnya.

---
## <h1 align = "center">DAFTAR PUSAKA <h1> 

[1] ITBOX by Course-Net. (2024). Pelajari dasar Python variable: Pengertian & contoh penulisannya. Diakses dari https://itbox.id/blog/pelajari-dasar-python-variable

[2] Jurnal Konversi Energi dan Manufaktur. (2023). Penerapan pemrograman Python dalam... Jurnal Konversi Energi dan Manufaktur, 8(1), 49–57. Diakses dari https://journal.unj.ac.id/unj/index.php/jkem/article/download/30368/14289

[3] Petani Kode. (2017). Belajar Python: Cara mengambil input dan menampilkan output. Diakses dari https://www.petanikode.com/python-input-output/

[4] RevoU. (2024). Belajar input di Python: Contoh dan tips mudah bagi pemula. Diakses dari https://www.revou.co/panduan-teknis/input-python

[5] UNMAHA. (2025). Memahami variabel dan tipe data di Python. Diakses dari https://blog.unmaha.ac.id/memahami-variabel-dan-tipe-data-di-python/